### PR TITLE
feat: added data platform support

### PR DESCRIPTION
**Pull request by iblindu**
## Checklist

- [x] PR name added as appropriate (e.g. `feat:`/`fix:`/`doc:`/`test:`/`refactor:`)
- [x] Tests added or updated
- [x] Documentation updated
- [ ] Sonar Cloud Scan
- [ ] Changelog updated and version incremented (label: upcoming release)
- [x] Github Issue linked if any
- [ ] Jira task updated

### CODE REVIEW

File: commands/dataplatform/examples.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package dataplatform
+
+const (
+	listClusterExample   = `ionosctl dataplatform cluster list`
+	getClusterExample    = `ionosctl dataplatform cluster get -i CLUSTER_ID`
+	createClusterExample = `ionosctl dataplatform cluster create --datacenter-id DATACENTER_ID --name NAME --version DATA_PLATFORM_VERSION
+
+ionosctl dbaas postgres cluster create -D DATACENTER_ID -L LAN_ID -C CIDR -U DB_USERNAME -P DB_PASSWORD`
```
Comment (amoga-ionos) [inline commands/dataplatform/examples.go:8]: I think this is a leftover from dbaas example

File: go.mod [hunk 2]
```diff
@@ -50,3 +51,7 @@ require (
 	gopkg.in/yaml.v2 v2.4.0 // indirect
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
 )
+
+require github.com/ionos-cloud/sdk-go-autoscaling v1.0.0
+
+replace github.com/ionos-cloud/sdk-go-autoscaling => ../../../../../sdk/stackable
```
Comment (amoga-ionos) [inline go.mod:57]: just a note to update this before merging

File: pkg/config/constants.go [hunk 3]
```diff
@@ -37,6 +37,11 @@ const (
 	ArgPassword            = "password"
 	ArgPasswordShort       = "p"
 	ArgNoHeaders           = "no-headers"
+	ArgFilters             = "filters"
+	ArgFiltersShort        = "F"
+	ArgOrderBy             = "order-by"
+	ArgMaxResults          = "max-results"
+	ArgMaxResultsShort     = "M"
```
Comment (amoga-ionos) [inline pkg/config/constants.go:44]: note to double check if these are used in here

**Review by amoga-ionos — APPROVED** [review]
Overall, this looks great. Small notes added:
- note to update `docs/summary.md` to link the docs generated
- I would recommend to use `goimports -w -l .` command to sort imports - this is more a nice to have to sort imports automatically.

:1st_place_medal:

**Comment by avirtopeanu-ionos** [comment]
https://github.com/ionos-cloud/ionosctl/pull/242

**Pull request closed without merging**